posthog-node/src/posthog-node.ts

```diff
@@ -101,39 +134,153 @@ export class PostHogGlobal implements PostHogNodeV1 {
   async getFeatureFlag(
     key: string,
     distinctId: string,
-    groups?: Record<string, string> | undefined
+    defaultResult?: boolean | string,
+    groups?: Record<string, string>,
+    personProperties?: Record<string, string>,
+    groupProperties?: Record<string, Record<string, string>>,
+    onlyEvaluateLocally: boolean = false,
+    sendFeatureFlagEvents: boolean = true
   ): Promise<string | boolean | undefined> {
```

Comment:
Not sure about this API tbh. I wonder if it isn't worth providing an object interface, that way it is much easier to add / rework properties in the future. 8 arguments for a function is definitely too many...

e.g. 😅 
`posthog.getFeatureFlag("flag", "id, false, null, { foo: "bar}, null, false, false)`

versus
```
posthog.getFeatureFlag("flag", "id", false, {
  personProperties: { foo: "bar" },
  sendFeatureFlags: false
}
```

Reply:
Thanks, was thinking the same. Tests became a bit too illegible with the current thing. I so miss keyword args :'(

The reason I stuck around was that when you want to set `onlyEvaluateLocally` to true, it's a good idea to have atleast some groups/person properties / etc. set, and the current API structure forces you to think about those params.

Will fix.